modules/siddhi-core/src/main/java/io/siddhi/core/table/CacheTable.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.core.table;
+
+import io.siddhi.core.config.SiddhiAppContext;
+import io.siddhi.core.config.SiddhiQueryContext;
+import io.siddhi.core.event.ComplexEvent;
+import io.siddhi.core.event.ComplexEventChunk;
+import io.siddhi.core.event.state.MetaStateEvent;
+import io.siddhi.core.event.state.StateEvent;
+import io.siddhi.core.event.stream.MetaStreamEvent;
+import io.siddhi.core.event.stream.StreamEvent;
+import io.siddhi.core.event.stream.StreamEventCloner;
+import io.siddhi.core.event.stream.StreamEventFactory;
+import io.siddhi.core.executor.VariableExpressionExecutor;
+import io.siddhi.core.table.holder.EventHolder;
+import io.siddhi.core.table.holder.IndexEventHolderForCache;
+import io.siddhi.core.table.record.RecordTableHandler;
+import io.siddhi.core.util.collection.AddingStreamEventExtractor;
+import io.siddhi.core.util.collection.operator.CompiledCondition;
+import io.siddhi.core.util.collection.operator.MatchingMetaInfoHolder;
+import io.siddhi.core.util.collection.operator.Operator;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.parser.EventHolderPasser;
+import io.siddhi.core.util.parser.OperatorParser;
+import io.siddhi.query.api.annotation.Annotation;
+import io.siddhi.query.api.annotation.Element;
+import io.siddhi.query.api.definition.Attribute;
+import io.siddhi.query.api.definition.TableDefinition;
+import io.siddhi.query.api.expression.Expression;
+import io.siddhi.query.api.expression.Variable;
+import io.siddhi.query.api.expression.condition.And;
+import io.siddhi.query.api.expression.condition.Compare;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.siddhi.query.api.util.AnnotationHelper.getAnnotation;
+
+/**
+ * common interface for FIFO, LRU, and LFU cache tables
+ */
+public abstract class CacheTable extends InMemoryTable {
+    private int maxSize;
+    private boolean cacheExpiryEnabled;
+    SiddhiAppContext siddhiAppContext;
+    int cachePolicyAttributePosition;
+    int numColumns;
+    int expiryAttributePosition;
+
```

Comment:
move following to the respective implimentation classes as privte variables. 
int cachePolicyAttributePosition;
int numColumns;
int expiryAttributePosition;